### PR TITLE
Added bxt_disable_skybox

### PR DESCRIPTION
**Pull request by SmileyAG**
Note: 

The main reason why I made this cvar is to solve #170 issue in an easy and dumb way, but if someone fixes it by modifying the wallhack or skybox-engine code, then I will close the PR probably (since I do not see any other useful use-cases, which require disabling the skybox in the game).

### CODE REVIEW

**Comment by chinese-soup** [comment]
1) Works on Linux.
2) If you wanna glClear then just add it to R_Clear like for example so:

```diff
diff --git a/BunnymodXT/modules/HwDLL.cpp b/BunnymodXT/modules/HwDLL.cpp
index cf5a7e9..50d01ac 100644
--- a/BunnymodXT/modules/HwDLL.cpp
+++ b/BunnymodXT/modules/HwDLL.cpp
@@ -4528,7 +4528,7 @@ HOOK_DEF_0(HwDLL, void, __cdecl, R_Clear)
 {
        // This is needed or everything will look washed out or with unintended
        // motion blur.
-       if (CVars::sv_cheats.GetBool() && CVars::bxt_wallhack.GetBool()) {
+       if ((CVars::sv_cheats.GetBool() && CVars::bxt_wallhack.GetBool()) || CVars::bxt_disable_skybox.GetBool()) {
                glClearColor(0.0f, 0.0f, 0.0f, 1.0f);
                glClear(GL_COLOR_BUFFER_BIT);
        }

```

3) Which ^ reminds me - maybe this should be gated behind sv_cheats 1 like bxt_wallhack is? Let's see what others think?

**Comment by SmileyAG** [comment]
>     2. If you wanna glClear then just add it to R_Clear like this for example:
> 
> 
> ```diff
> diff --git a/BunnymodXT/modules/HwDLL.cpp b/BunnymodXT/modules/HwDLL.cpp
> index cf5a7e9..50d01ac 100644
> --- a/BunnymodXT/modules/HwDLL.cpp
> +++ b/BunnymodXT/modules/HwDLL.cpp
> @@ -4528,7 +4528,7 @@ HOOK_DEF_0(HwDLL, void, __cdecl, R_Clear)
>  {
>         // This is needed or everything will look washed out or with unintended
>         // motion blur.
> -       if (CVars::sv_cheats.GetBool() && CVars::bxt_wallhack.GetBool()) {
> +       if ((CVars::sv_cheats.GetBool() && CVars::bxt_wallhack.GetBool()) || CVars::bxt_disable_skybox.GetBool()) {
>                 glClearColor(0.0f, 0.0f, 0.0f, 1.0f);
>                 glClear(GL_COLOR_BUFFER_BIT);
>         }
> ```

Done. Thanks!

File: BunnymodXT/modules/HwDLL.cpp [hunk 1]
```diff
@@ -4285,6 +4305,14 @@ HOOK_DEF_0(HwDLL, int, __cdecl, V_FadeAlpha)
 		return ORIG_V_FadeAlpha();
 }
 
+HOOK_DEF_0(HwDLL, void, __cdecl, R_DrawSkyBox)
+{
+	if (CVars::bxt_disable_skybox.GetBool())
+		return;
+	else
+		return ORIG_R_DrawSkyBox();
```
Comment (hobokenn) [inline BunnymodXT/modules/HwDLL.cpp:4313]:
```suggestion
	if (CVars::sv_cheats.GetBool() && CVars::bxt_disable_skybox.GetBool())
		return;

	ORIG_R_DrawSkyBox();
```

File: BunnymodXT/modules/HwDLL.cpp [hunk 2]
```diff
@@ -4502,7 +4530,7 @@ HOOK_DEF_0(HwDLL, void, __cdecl, R_Clear)
 {
 	// This is needed or everything will look washed out or with unintended
 	// motion blur.
-	if (CVars::sv_cheats.GetBool() && CVars::bxt_wallhack.GetBool()) {
+	if (CVars::sv_cheats.GetBool() && CVars::bxt_wallhack.GetBool() || CVars::bxt_disable_skybox.GetBool()) {
```
Comment (hobokenn) [inline BunnymodXT/modules/HwDLL.cpp:4533]:
```suggestion
	if (CVars::sv_cheats.GetBool() && (CVars::bxt_wallhack.GetBool() || CVars::bxt_disable_skybox.GetBool())) {
```

File: BunnymodXT/patterns.hpp [hunk 3]
```diff
@@ -340,6 +340,11 @@ namespace patterns
 			"HL-NGHL",
 			"B8 28 54 01 00 E8 ?? ?? ?? ?? 8B 0D ?? ?? ?? ?? 53 8B 9C 24 30 54 01 00"
 		);
+
+		PATTERNS(R_DrawSkyBox,
+			"HL-SteamPipe",
+			"55 8B EC 83 EC 1C A1 ?? ?? ?? ?? 53 56 BB 00 00 80 3F 57 C7 45 F8 00 00 00 00 85 C0"
```
Comment (hobokenn) [inline BunnymodXT/patterns.hpp:346]:
```suggestion
			"55 8B EC 83 EC 1C A1 ?? ?? ?? ?? 53 56",
			"HL-4554",
			"83 EC 1C A1 ?? ?? ?? ?? 53 55",
			"HL-WON-1712",
			"83 EC 0C 53 55 56 57 E8 ?? ?? ?? ?? 33 FF"
```

**Review by hobokenn — CHANGES_REQUESTED** [review]

**Comment by YaLTeR** [comment]
Thanks

**Pull request merged**